component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -546,6 +547,20 @@ private void validateUsingSubject(String subject, AuthenticationContext authenti
         }
     }
 
+    /**
+     * Check the authenticated user's tenant domain and verify whether it from super tenant or different tenant.
+     *
+     * @param authenticatedUser Get the authenticated user object from the authentication context  .
+     */
+    private String getAuthenticatedUserName(AuthenticatedUser authenticatedUser) {
+
+        String carbonSuperUser = null;
+        if (authenticatedUser.getAuthenticatedSubjectIdentifier().contains("carbon.super")) {
```

Comment:
Assign "authenticatedUser.getAuthenticatedSubjectIdentifier()" to a variable and reuse it. better to use **endswith** than **contains**

Reply:
Use the constant MultitenantConstants.SUPER_TENANT_DOMAIN_NAME instead of "carbon.super"